File: python/parmana/api/replay_api.py
"""
Parmana Replay API.

Replay Business Transactions.
"""

from __future__ import annotations

from parmana.config.transport import Transport
from parmana.models.replay_result import ReplayResult


class ReplayApi:
    """
    Replay API.

    Responsibilities
    ----------------
    - Replay Business Transactions
    - Verify deterministic execution

    This API does NOT:
    - execute Business Transactions
    - verify trust records
    - generate receipts
    - validate policies
    """

    def __init__(
        self,
        transport: Transport,
    ) -> None:
        self._transport = transport

    def replay(
        self,
        business_transaction_id: str,
    ) -> ReplayResult:
        """
        Replay a Business Transaction.

        Parameters
        ----------
        business_transaction_id:
            Business Transaction identifier.

        Returns
        -------
        Replay Result.
        """

        return self._transport.send(
            method="POST",
            path="/replay",
            body={
                "businessTransactionId": business_transaction_id,
            },
            response_model=ReplayResult,
        )

File: typescript/src/client/ReplayApi.ts
/**
 * Parmana TypeScript SDK
 *
 * Replay API.
 *
 * Replay Business Transactions.
 */

import type {
  Transport,
} from "../config/Transport.js";

import type {
  ReplayResult,
} from "../models/replay-result.js";

/**
 * Replay API.
 */
export class ReplayApi {
  constructor(
    private readonly transport: Transport,
  ) {}

  /**
   * Replay a Business Transaction.
   */
  public async replay(
    businessTransactionId: string,
  ): Promise<ReplayResult> {
    const response =
      await this.transport.send<ReplayResult>({
        method: "POST",
        path: "/replay",
        body: {
          businessTransactionId,
        },
      });

    return response.body;
  }
}